docs/book/src/clusterctl/init.md

```diff
@@ -0,0 +1,233 @@
+# clusterctl init
+
+The `clusterctl init` command installs the cluster-API components and transforms the Kubernetes cluster
+into a management cluster.
+
+This document provides more detail on how `clusterctl init` works and on the supported options for customizing your
+management cluster.
+
+## Defining the management cluster 
+
+The `clusterctl init` command accepts in input a list of providers to install.
+
+<aside class="note">
+
+<h1> Which providers can I use? </h1>
+
+You can use the `clusterctl config providers` to get the list of supported providers.
+ 
+If the provider of your choice is missing, you can customize the list of supported providers by using the
+[clusterctl configuration](configuring-clusterctl.md) file. 
+
+</aside>
+
+#### Automatically installed providers
+
+The `clusterctl init` command automatically adds the Cluster API core provider and
+the kubeadm bootstrap provider to the list of providers to install. This allows users to use a concise command syntax for initializing a management cluster. e.g.
```

Comment:
I still have to tackle the kubeadm control plane and its implications (not only for init, but also for templates)
ATM I assume the kubeadm control plane is optional, so I'm leaning toward not install by default because otherwise we should introduce a way to let the user to opt-out from installing it. But I need to talk with @detiber / @vincepri to clarify expectations/implications

Reply:
We definitely want people using KubeadmControlPlane to provision the control plane for a cluster. I think it should be installed by default.